src/BloomExe/Edit/PageListView.cs

```diff
@@ -142,11 +142,18 @@ public new bool Enabled
 			set { _thumbNailList.Enabled = value; }
 		}
 
+		private bool _processingClickHandler = false;
 		private void _addPageButton_Click(object sender, EventArgs e)
 		{
+			if (_processingClickHandler)
+				return;
+			_processingClickHandler = true;
```

Comment:
Should this be in a try...finally block to ensure that _processingClickHandler always gets reset to false at the end?  (I tend to be somewhat paranoid about such things.)